src/stemmabench/variant_analyzer.py

```diff
@@ -0,0 +1,703 @@
+        diagonal consists of k-D arrays filled only with `False` because each 
+        witness is obvisouly fully similar to itself, reading by reading.
+
+        Returns:
+            np.ndarray: A 3-D (n, n, k) Boolean matrix indicating variant 
+                locations among witnesses.
+        """
+        return np.expand_dims(self.array, axis=0) != np.expand_dims(self.array, axis=1)
+
+    # OMIT
+    @staticmethod
+    def is_omit(word1: str, word2: str, missing: str = "-") -> bool:
+        """
+        Check if two words represent an omission based on a specified missing 
+        character.
+
+        Args:
+            word1 (str): The first word.
+            word2 (str): The second word.
+            missing (str, optional): The character representing missing 
+                readings. Defaults to "-".
+
+        Returns:
+            bool: True if the words represent an omission, False otherwise.
+        """
+        
+        return word1 != word2 and (word1 == missing or word2 == missing)
+    
+    # MISPELL
+    @staticmethod
+    def is_mispell(word1: str, 
+                   word2: str, 
+                   distance: str = "DamerauLevenshtein", 
+                   mispell_cutoff: float = 0.4) -> bool:
+        """
+        Check if two words are considered a misspelling based on a specified 
+        distance metric and cutoff score.
+
+        Args:
+            word1 (str): The first word.
+            word2 (str): The second word.
+            distance (str, optional): The distance metric to use.
+                (default: "DamerauLevenshtein").
+                - "Levenshtein"
+                - "DamerauLevenshtein"
+                - "Hamming"
+                - "JaroWinkler"
+            mispell_cutoff (float, optional): The cutoff score for considering 
+                two words as a misspelling (default: 0.4).
+
+        Returns:
+            bool: True if the words are a misspelling, False otherwise.
+        
+        Raises:
+            ValueError: If an unsupported distance metric is provided.
+        """
+        dist_funcs = {
+            "Levenshtein": textdistance.Levenshtein().normalized_distance,
+            "DamerauLevenshtein": (textdistance.DamerauLevenshtein()
+                                   .normalized_distance),
+            "Hamming": textdistance.Hamming().normalized_distance,
+            "JaroWinkler": textdistance.JaroWinkler().normalized_distance
+        }
+        if distance not in dist_funcs.keys():
+            raise ValueError(f"Distance {distance} is not supported. "
+                             f"Choose one among {dist_funcs.keys()}")
+        norm_dist = dist_funcs[distance](str(word1), str(word2))
+        # Mispell if distance > 0 (not exact match) 
+        # but distance < cutoff (not too different)
+        return bool(0 < norm_dist <= mispell_cutoff)
+    
+    # SYNONYMS
+    @staticmethod
+    def synonyms(word: str) -> set[str]:
```

Comment:
shouldn't this depend on the used language ?

Reply:
I have added an if-clause but it will need to be more flexible.